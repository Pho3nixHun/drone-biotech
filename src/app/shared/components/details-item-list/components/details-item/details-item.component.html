<div class="flex flex-row space-x-4">
    <div><ng-content select="app-avatar" /></div>
    <div>
        <ng-content select="[h4, p]" />
    </div>
</div>
