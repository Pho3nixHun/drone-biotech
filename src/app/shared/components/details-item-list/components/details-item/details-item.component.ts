import { Component } from '@angular/core';

@Component({
    selector: 'app-details-item',
    imports: [],
    templateUrl: './details-item.component.html',
})
export class DetailsItemComponent {}
