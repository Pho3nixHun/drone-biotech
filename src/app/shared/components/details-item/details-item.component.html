<div class="flex flex-row space-x-4">
    <div><ng-content select="div" /></div>
    <div>
        <ng-content select="[h3, p]" />
    </div>
</div>
