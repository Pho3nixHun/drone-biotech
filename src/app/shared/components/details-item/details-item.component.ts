import { Component } from '@angular/core';

/**
 * DetailsItemComponent
 *
 * Type: Presentational
 *
 * Scope:
 * - Responsible for rendering an individual details item.
 *
 * Out-of-Scope:
 * - Should not handle data fetching.
 * - Contains no business logic related to the item.
 * - Does not manage or transform the item's data.
 *
 * Purpose (optional):
 * To visually present an individual details item.
 */
@Component({
    selector: 'app-details-item',
    imports: [],
    templateUrl: './details-item.component.html',
})
export class DetailsItemComponent {}
