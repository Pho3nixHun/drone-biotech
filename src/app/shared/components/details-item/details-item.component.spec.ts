import { ComponentFixture, TestBed } from '@angular/core/testing';
import { DetailsItemComponent } from './details-item.component';
import { Component } from '@angular/core';

@Component({
    imports: [DetailsItemComponent],
    template: `
        <app-details-item>
            <div>Should be projected</div>
            <h3>Should be projected</h3>
            <p>Should be projected</p>
            <time datetime="">Should not be projected</time>
        </app-details-item>
    `,
})
class TestHostComponent {}

describe('DetailsItemComponent', () => {
    let fixture: ComponentFixture<TestHostComponent>;
    let compiled: HTMLElement;

    beforeEach(async () => {
        await TestBed.configureTestingModule({
            imports: [TestHostComponent],
        }).compileComponents();

        fixture = TestBed.createComponent(TestHostComponent);
        compiled = fixture.debugElement.nativeElement;
    });

    // Snapshot testing
    it('should project the content and render the template correctly', () => {
        // There is no need to arrange

        // Act
        fixture.detectChanges();

        // Assert
        expect(compiled).toMatchSnapshot();
    });
});
