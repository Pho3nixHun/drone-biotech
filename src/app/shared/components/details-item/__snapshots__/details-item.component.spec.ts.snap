// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsItemComponent should project the content and render the template correctly 1`] = `
<div
  id="root0"
  ng-version="19.2.9"
>
  <app-details-item>
    <div
      class="flex flex-row space-x-4"
    >
      <div>
        <div>
          Should be projected
        </div>
      </div>
      <div>
        <h3>
          Should be projected
        </h3>
        <p>
          Should be projected
        </p>
      </div>
    </div>
  </app-details-item>
</div>
`;
